### PR TITLE
[match] Add integration tests for match service

### PR DESCRIPTION
**Pull request by jaylees14**
- Add integration tests to match service
- Fix issue with error messages not displaying correctly 

Is that a 150 line test you ask?

### CODE REVIEW

File: match/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - PGUSER=postgres
+    volumes:
+      - ./../user-db/init.sql:/docker-entrypoint-initdb.d/init.sql
+    networks:
+      - user-network
+
+networks:
+  user-network:
+  match-network:
+  parent-network:
```
Comment (c-harding) [inline match/docker-compose.yml]: Not a fan of the lack of trailing newline

File: match/match_it_test.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	req.Header.Set("Authorization", "Bearer "+JWT1)
+	_, err = client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestIntegrationMatch(t *testing.T) {
```
Comment (c-harding) [inline match/match_it_test.go:67]: Might be worth thinking about using private tests as part of the mega test, though not sure how well that will work with logging of lines etc

    func TestIntegrationMega() {
        setUpStuff...
        testPartOne();
        testPartTwo();
        testPartThree();
    }

**Review by c-harding — COMMENTED** [review]
If the tests pass that’s good, but there’s too much stuff to fully review now

File: match/match_it_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req, err = http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(`{"Name": "Lewis"}`)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Set("Authorization", "Bearer "+JWT1)
```
Comment (c-harding) [inline match/match_it_test.go]: `+` used here, Sprintf used elsewhere

File: match/match_it_test.go [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+	_, err = time.Parse(time.RFC3339, createMatchResponse.MatchedOn)
+	if err != nil {
+		t.Fatalf("MatchedOn was in an invalid format: %s", err.Error())
+	}
+
+	return createMatchResponse.ID
+}
+
+func testReadMatch(t *testing.T, uuid uuid.UUID) {
+	res, err := makeRequest(environment, http.MethodGet, fmt.Sprintf("/match/%s", uuid.String()), "", JWT0)
```
Comment (c-harding) [inline match/match_it_test.go:109]: Sprintf used here, `+` used elsewhere

File: match/match_it_test.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+func testCreateMatch(t *testing.T) uuid.UUID {
+	// Create match
+	res, err := makeRequest(environment, http.MethodPost, "/match", fmt.Sprintf(`{"UserOne": "%s", "UserTwo": "%s"}`, UUID0, UUID1), JWT0)
+	if err != nil {
+		t.Fatalf("Could not make request: %s", err.Error())
+	}
+
+	if res.Code != http.StatusOK {
+		t.Errorf("Wrong status code: %v", res.Code)
```
Comment (willjvsmith) [inline match/match_it_test.go]: Probs error -> fatal

File: match/match_it_test.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+	os.Exit(m.Run())
+}
+
+func TestIntegrationMatch(t *testing.T) {
+	id := testCreateMatch(t)
+	testReadMatch(t, id)
+	testReadMatchList(t)
+	testUpdateMatch(t, id)
+	testDeleteMatch(t, id)
```
Comment (c-harding) [inline match/match_it_test.go:72]: So legible ❤️

**Review by willjvsmith — APPROVED** [review]

**Review by c-harding — APPROVED** [review]

**Pull request merged**